### PR TITLE
Add Support for Entities/Chunks/Worlds/Blocks in CondIsWithin

### PR DESCRIPTION
**Pull request by sovdeeth**
### Description

Modifies CondIsWithinLocation to support checking if a location is within an entity's hitbox, a block's collision shape (1.17+), a chunk, or a world.
Also removes CondIsInWorld in favor of the new syntax in CondIsWithin

Possible breaking change: `%entity% is in [[the] world] %world%` ->`%locations% (is|are) (within|in[side [of]]) %world%`

---
**Target Minecraft Versions:** any 
**Requirements:** 1.17+ for within blocks, any otherwise 
**Related Issues:**

### CODE REVIEW

**Review by UnderscoreTud — CHANGES_REQUESTED** [review]

File: src/main/java/ch/njol/skript/conditions/CondIsWithin.java [hunk 1]
```diff
@@ -0,0 +1,157 @@
...
+	"",
+	"if player is in world(\"world\"):",
+		"\tsend \"You are in the overworld!\" to player",
+	"",
+	"if attacker's location is inside of victim:",
+		"\tcancel event",
+		"\tsend \"Back up!\" to attacker and victim",
+})
+@Since("INSERT VERSION")
+@RequiredPlugins("1.17 or newer (for within block)")
```
Comment (AyhamAl-Ali) [inline src/main/java/ch/njol/skript/conditions/CondIsWithin.java:58]:
```suggestion
@RequiredPlugins("MC 1.17+ (within block)")
```

File: src/test/skript/tests/syntaxes/conditions/CondIsWithin.sk [hunk 2]
```diff
@@ -0,0 +1,56 @@
...
+	assert location(10, 10, 10, "world") is within world("world") with "failed within world ##1"
+
+	# entities
+	set {_loc} to spawn of world "world"
+	spawn a pig at {_loc}
+	set {_pig} to last spawned entity
+	assert {_loc} is within {_pig} with "failed within entity ##1"
+	assert {_loc1} is not within {_pig} with "failed within entity ##2"
+
+	delete random entity of {_pig}
```
Comment (AyhamAl-Ali) [inline src/test/skript/tests/syntaxes/conditions/CondIsWithin.sk:56]:
```suggestion
	delete random entity of {_pig}
	
```

**Review by AyhamAl-Ali — CHANGES_REQUESTED** [review]
Very nice PR ⚡ Well done

File: src/main/java/ch/njol/skript/conditions/CondIsWithin.java [hunk 3]
```diff
@@ -0,0 +1,157 @@
...
+ *  You should have received a copy of the GNU General Public License
+ *  along with Skript.  If not, see <http://www.gnu.org/licenses/>.
+ *
+ * Copyright Peter Güttinger, SkriptLang team and contributors
+ */
+package ch.njol.skript.conditions;
+
+import ch.njol.skript.Skript;
+import ch.njol.skript.conditions.base.PropertyCondition;
+import ch.njol.skript.doc.*;
```
[the remark this answers is not in the document]
Comment (sovdeeth) [reply src/main/java/ch/njol/skript/conditions/CondIsWithin.java:23]: Oh my god intellij i set the limit to like 50 imports before stars should be used
I'm going to murder something

**Review by AyhamAl-Ali — APPROVED** [review]
Code wise looks clean, I haven't tested it tho hopefully it works 😄

File: src/main/java/ch/njol/skript/conditions/CondIsWithin.java [hunk 4]
```diff
@@ -0,0 +1,160 @@
...
+		locsToCheck = (Expression<Location>) exprs[0];
+		if (matchedPattern <= 1) {
+			// within two locations
+			withinLocations = true;
+			loc1 = (Expression<Location>) exprs[1];
+			loc2 = (Expression<Location>) exprs[2];
+		} else {
+			// within an entity/block/chunk/world
+			withinLocations = false;
+			area = (Expression<Object>) exprs[1];
```
Comment (UnderscoreTud) [inline src/main/java/ch/njol/skript/conditions/CondIsWithin.java:93]:
```suggestion
			area = exprs[1];
```

**Review by UnderscoreTud — APPROVED** [review]

File: src/main/java/ch/njol/skript/conditions/CondIsWithin.java (hunk 4 above)
Comment (sovdeeth) [reply src/main/java/ch/njol/skript/conditions/CondIsWithin.java:93]: Oh good catch, I'll fix shortly :)

File: src/main/java/ch/njol/skript/conditions/CondIsWithin.java [hunk 5]
```diff
@@ -0,0 +1,160 @@
...
+	static {
+		String validTypes = "entity/chunk/world";
+		if (Skript.methodExists(Block.class, "getCollisionShape"))
+			validTypes += "/block";
+
+		Skript.registerCondition(CondIsWithin.class,
+			"%locations% (is|are) within %location% and %location%",
+			"%locations% (isn't|is not|aren't|are not) within %location% and %location%",
+			"%locations% (is|are) (within|in[side [of]]) %" + validTypes + "%",
+			"%locations% (isn't|is not|aren't|are not) (within|in[side [of]]) %" + validTypes + "%");
```
Comment (TheLimeGlass) [inline src/main/java/ch/njol/skript/conditions/CondIsWithin.java:73]:
```suggestion
				"%locations% (is|are) within %location% and %location%",
				"%locations% (isn't|is not|aren't|are not) within %location% and %location%",
				"%locations% (is|are) (within|in[side [of]]) %" + validTypes + "%",
				"%locations% (isn't|is not|aren't|are not) (within|in[side [of]]) %" + validTypes + "%"
		);
```

**Review by TheLimeGlass — CHANGES_REQUESTED** [review]

**Review by TheLimeGlass — APPROVED** [review]

**Pull request merged**